src/post.ts

```diff
@@ -1,20 +1,32 @@
 import {EventPayload} from './EventPayload';
 
 /**
- * Used by the AnalyticsEventReport report() method to send an analytics event request via the
- * Beacon API.
- *
- * todo: Update this method to default to using fetch + keepalive unless not supported and
- * forceFetch is not set to true. Also update the description above.
+ * Used by the AnalyticsEventReport report() method to send an analytics event request via
+ * fetch, with the keepalive option set to true. If forceFetch is false and keepalive is not
+ * supported by the browser i.e. Firefox, the request is sent via the Beacon API.
  *
  * @param url URL that the request will be sent to
  * @param body the EventPayload object
+ * @param forceFetch if true, it will force the browser to call fetch instead of sendBeacon,
+ * regardless of whether keepAlive is supported
  */
-const post = (
-    url: string,
-    body: EventPayload
-) => {
-    return navigator.sendBeacon(url, JSON.stringify(body));
+export function post(url: string, body: EventPayload, forceFetch: boolean) {
+    if (!forceFetch && useBeacon(body)) {
+        return navigator.sendBeacon(url, JSON.stringify(body));
+    }
+    return fetch(url, {method: 'POST', body: JSON.stringify(body), keepalive: true});
 }
 
-export default post;
+/**
+ * Returns a boolean that determines whether the post method should use beacon. Firefox and Firefox
+ * for Android do not support fetch + keepAlive, so useBeacon will return true if the browser is
+ * Firefox or Firefox for Android.
+ *
+ * @param body the EventPayload object
+ */
+export function useBeacon(body: EventPayload) {
+    const agent = body.browserAgent;
```

Comment:
Should we window.navigator.userAgent? Or are we assuming this will be set from the report function? Someone needs to set the browserAgent somewhere.

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgent

Reply:
The useBeacon method checks for either the window.navigator.userAgent or the user-set browserAgent.browser. Would you suggest just using window.navigator.userAgent and ignoring the user-set value since we don't want it to override the window.navigator.userAgent?

Reply:
My original question for this was what cases would we want to be using the user set parameter to figure out if we should use beacon or fetch.

The main reason we need this check is because we want to make sure we are using the right function for the given browser right? Does the user set one tell us anything about the current browser?

Reply:
Hmm, don't think the user set parameter is reliable for checking the current browser and shouldn't take precedence over the window.navigator.userAgent. Removed the check for the user set parameter.